src/core/api/video_track_manager.ts

```diff
@@ -0,0 +1,389 @@
+      id === wantedId);
+
+    if (wantedAdaptation === undefined) {
+      throw new Error("Video Track not found.");
+    }
+    const chosenVideoAdaptation = this._videoChoiceMemory.get(period);
+    if (chosenVideoAdaptation === wantedAdaptation) {
+      return;
+    }
+
+    this._videoChoiceMemory.set(period, wantedAdaptation);
+    videoInfos.adaptation$.next(wantedAdaptation);
+  }
+
+  /**
+   * Disable the given video track for a given Period.
+   *
+   * @param {Period} period - The concerned Period.
+   *
+   * @throws Error - Throws if the period given has not been added
+   */
+  public disableVideoTrack(period : Period) : void {
+    const periodItem = getPeriodItem(this._periods, period);
+    const videoInfos = periodItem && periodItem.video;
+    if (!videoInfos) {
+      throw new Error("LanguageManager: Given Period not found.");
+    }
+    const chosenVideoAdaptation = this._videoChoiceMemory.get(period);
+    if (chosenVideoAdaptation === null) {
+      return;
+    }
+
+    this._videoChoiceMemory.set(period, null);
+    videoInfos.adaptation$.next(null);
+  }
+
+  /**
+   * Returns an object describing the chosen video track for the given video
+   * Period.
+   *
+   * Returns null is the the current video track is disabled or not
+   * set yet.
+   *
+   * @param {Period} period
+   * @returns {Object|null}
+   */
+  public getChosenVideoTrack(period : Period) : ILMVideoTrack|null {
+    const periodItem = getPeriodItem(this._periods, period);
+    const videoInfos = periodItem && periodItem.video;
+    if (videoInfos == null) {
+      return null;
+    }
+
+    const chosenVideoAdaptation = this._videoChoiceMemory.get(period);
+    if (!chosenVideoAdaptation) {
+      return null;
+    }
+
+    const representations = chosenVideoAdaptation
+      .representations
+      .map((representation) => {
+        return {
+          id: representation.id,
+          bitrate: representation.bitrate,
+          frameRate: representation.frameRate,
+          width: representation.width,
+          height: representation.height,
+          codec: representation.codec,
+        };
+      });
+
+    return {
+      id: chosenVideoAdaptation.id,
+      representations,
+    };
+  }
+
+  /**
+   * Returns all available video tracks for a given Period, as an array of
+   * objects.
+   *
+   * @returns {Array.<Object>}
+   */
+  public getAvailableVideoTracks(period : Period) : ILMVideoTrackListItem[] {
+    const periodItem = getPeriodItem(this._periods, period);
+    const videoInfos = periodItem && periodItem.video;
+    if (videoInfos == null) {
+      return [];
+    }
+
+    const chosenVideoAdaptation = this._videoChoiceMemory.get(period);
+    const currentId = chosenVideoAdaptation && chosenVideoAdaptation.id;
+
+    return videoInfos.adaptations
+      .map((adaptation) => {
+
+        const representations = adaptation
+          .representations
+          .map((representation) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 309. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=285&issues=AWU4an4qW2AOP0CkHpBt&open=AWU4an4qW2AOP0CkHpBt)

[ISSUE_KEY]: AWU4an4qW2AOP0CkHpBt